src/main/java/org/jboss/tm/usertx/TransactionListener.java

```diff
@@ -0,0 +1,16 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.Transaction;
+
+/**
+ * A listener callback for transaction related events
+ */
+public interface TransactionListener {
+    /**
+     * The callback for notifying registered listeners of a transaction related event
+     *
+     * @param transaction the transaction that the caller is registering interest in
+     * @param transactionEvent indication of what kind of change has occurred
+     */
+    void stateChange(Transaction transaction, TransactionEvent transactionEvent);
```

Comment:
The advantage of not using Transaction in the interface is that it could be provided with a ControlWrapper in the future if we wanted to update TxAssociations to use something close to this API.